### PR TITLE
LF-4316: Product details remain after removing product in complete flow

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

It looks like `undefined` passed to `useFieldArray`'s `append()` is overridden by the default value. It works as expected by changing it to an empty string.

Jira link: https://lite-farm.atlassian.net/browse/LF-4316

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Comment by Duncan-Brain** [comment]
I am still seeing the Quantity be pre-populated with 0:
Screenshot 2024-07-22 at 3 54 06 PM

Also noticed my uniqueness check returned an error 😓 .. I missed a case that requires previously made controller function.

**Comment by SayakaOno** [comment]
Thank you @Duncan-Brain, I missed the "0"...
What's the uniqueness check? Is it something that needs to be covered in this PR?

File: packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx [hunk 1]
```diff
@@ -42,9 +42,21 @@ interface ProductFields {
...
   [TASK_PRODUCT_FIELD_NAMES.IS_WEIGHT]: true,
+  [TASK_PRODUCT_FIELD_NAMES.WEIGHT]: NaN,
+  [TASK_PRODUCT_FIELD_NAMES.WEIGHT_UNIT]: undefined,
+  [TASK_PRODUCT_FIELD_NAMES.APPLICATION_RATE_WEIGHT]: NaN,
+  [TASK_PRODUCT_FIELD_NAMES.APPLICATION_RATE_WEIGHT_UNIT]: undefined,
+  [TASK_PRODUCT_FIELD_NAMES.VOLUME]: NaN,
+  [TASK_PRODUCT_FIELD_NAMES.VOLUME_UNIT]: undefined,
+  [TASK_PRODUCT_FIELD_NAMES.APPLICATION_RATE_VOLUME]: NaN,
+  [TASK_PRODUCT_FIELD_NAMES.APPLICATION_RATE_VOLUME_UNIT]: undefined,
 };
```
Comment (SayakaOno) [inline packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx:60]: I believe defining defultValues for all the fields resolves "A component is changing an uncontrolled input to be controlled." warning, which is nice.

**Review by kathyavini — APPROVED** [review]
Looks good, and I am really enjoying not seeing that uncontrolled to controlled input warning!!

**Review by Duncan-Brain — APPROVED** [review]
Looks amazing! Awesome about the console error too!

Nothing you need to do regarding my uniqueness error - now that select options are fixed I just noticed it now when I should have before!

**Pull request merged**